### DISCUSSION TITLE
Service key client gets user not allowed response

### DISCUSSION THREAD

**Discussion by pnilssson** [supabase/supabase#10418] [Questions] [answered]
I'm trying to use my supabase client server side initiated with the service key. When I log the supabase client I can see that the apiKey of the GoTrueAdminApi object has the correct key. I can see that the SupabaseClient object has the correct subabaseKey. Yet I get "User not allowed" error response when trying to use the `auth.admin.updateUserById` method. Has anyone else has similar problems and can point me in the right direction? I really don't know what would cause this issue.

**Comment by pnilssson** [comment]
I just found this now: https://github.com/supabase/supabase/discussions/9314#discussioncomment-4105780
Would have something to do with it? If so the documentation isn’t reliable currently as it says that you should be able to use the auth.admin methods with the service key.

**Comment by GaryAustin1** [comment, marked as answer]
You are somewhat unclear on what keys you are seeing in the objects.  The authorization header is what must be set to the service_role key.  Normally if you don't have a user that header defaults to the apikey/supabaseKey you initialized with.  But if you set up a user or replace the authorization header with another token then the service_role will not be passed thru.

**Reply by pnilssson** [reply]
My setup was that I had one supabase client(let's call this **client 1**) where I did:
`  await supabase.auth.setSession({
    refresh_token: refreshToken,
    access_token: accessToken,
  })`
early in my application. Therefore I couldn't use it to do admin methods with just as you said. I'm still a bit confused though. Using **client 1** I'm able to do `await supabase.auth.getUser()` to get user information but I'm not able to do `await supabase.auth.updateUser({password: 'new password'})` to update the user, I get "No session found" error.

So now I'm using **client 1** to get the user id using `await supabase.auth.getUser()`. I then create a new client(**client 2**) initiated with the service key and use that to `await supabase.auth.admin.updateUserById('<user id I got from client 1>', { password: 'new_password' })`
Does that makes sense? Or is there a better way to solve this?

**Reply by lacykerel** [reply]
I'm having a similar issue where documentation and walk throughs define having two clients 
`export const supabase = createClient(supabaseUrl, supabaseKey);
export const supabaseAdmin = createClient(supabaseUrl, serviceKey);
`

however when setup this way the supabaseAdmin const createClient says it needs the other key still.

**Reply by pnilssson** [reply]
What kind of error message do you get? And what kind of method are you trying to do using the supabaseAdmin when you get this error?
